npm/pkg/dataplane/ipsets/dirtycache_linux.go

```diff
@@ -0,0 +1,119 @@
+package ipsets
+
+import "fmt"
+
+type memberAwareDirtyCache struct {
+	// map of prefixed set names to original members
+	toAddOrUpdateCache map[string]map[string]struct{}
```

Comment:
instead of maintaining all original members, i think saving just the members to be added and members to be deleted is memory friendly. Because take the example of recent scale testing we have done, if an IPSET has large member count and we just need to a single pod ip from it, we are duplicating the large member set into dirtycache .